array/numeric.gen.go

```diff
@@ -0,0 +1,174 @@
+// Generated by tmpl
+// https://github.com/benbjohnson/tmpl
+//
+// DO NOT EDIT!
+// Source: numeric.gen.go.tmpl
+
+package array
+
+import (
+	"github.com/apache/arrow/go/arrow/array"
+	"github.com/apache/arrow/go/arrow/memory"
+)
+
+type (
+	Int   = array.Int64
+	Uint  = array.Uint64
+	Float = array.Float64
+)
+
+type IntBuilder struct {
+	b *array.Int64Builder
```

Comment:
I guess you could have omitted the field name `b` here and then you would have gotten all of the methods from the embedded builder. Why not just do that?

Reply:
I intentionally did this so that we don't automatically expose the embedded builder methods. I'd like to keep our API explicit. When you embed a type, you add a public field to the struct that becomes part of the API and now can't be removed without breaking API compatibility. The lowercase field attribute keeps the fields in this struct unexported and able to be changed for an internal representation.